kale/prepdata/supergraph_construct.py

```diff
@@ -0,0 +1,250 @@
+        >>> supervertex_weight2 = SuperVertex(node_feat, edge_index, edge_type, edge_weight)
+    """
+
+    def __init__(
+        self,
+        name: str,
+        node_feat: torch.Tensor,
+        edge_index: torch.Tensor,
+        edge_type: torch.Tensor = None,
+        edge_weight: torch.Tensor = None,
+    ) -> None:
+
+        self.name = name
+        self.node_feat = node_feat
+        self.edge_index = edge_index
+        self.edge_type = edge_type
+        self.edge_weight = edge_weight
+
+        # get the number of nodes, node features and edges
+        self.num_node, self.num_node_feat = node_feat.shape
+        self.num_edge = edge_index.shape[1]
+
+        # initialize in-supervertex and out-supervertex lists
+        self.in_supervertex_list: List[str] = []
+        self.out_supervertex_list: List[str] = []
+        self.if_start_supervertex = True
+
+        self.__process_edges__()
+
+    def __process_edges__(self):
+        r"""
+        process the edges of the supervertex.
+        """
+        # get the number of edge types
+        if self.edge_type is None:
+            self.num_edge_type = 1
+        else:
+            unique_edge_type = self.edge_type.unique()
+            self.num_edge_type = unique_edge_type.shape[0]
+
+            # check if the index of edge type is continuous and starts from 0
+            if self.num_edge_type != unique_edge_type.max() + 1:
+                error_msg = "The index of edge type is not continuous and starts from 0."
+                logging.error(error_msg)
+                raise ValueError(error_msg)
+
+            # sort the edges and edge types
+            sort_index = torch.argsort(self.edge_type)
+            self.edge_index = self.edge_index[:, sort_index]
+            self.edge_type = self.edge_type[sort_index]
+
+            self.__get_range_list__()
+
+    def __get_range_list__(self):
+        """get the range of edge types"""
+        idx = 0
+        range_list = [[0, 0]]
+
+        for i, et in enumerate(self.edge_type):
+            if et != idx:
+                idx = et
+                range_list[-1][1] = i
+                range_list.append([i, i])
+
+        range_list[-1][1] = i + 1
+
+        self.range_list = torch.tensor(range_list)
+
+    def __repr__(self) -> str:
+        return f"SuperVertex(\n    name={self.name}, \n    node_feat={self.node_feat.shape}, \n    edge_index={self.edge_index.shape}, \n    num_edge_type={self.num_edge_type})"
+
+    def add_in_supervertex(self, vertex_name: str):
+        self.in_supervertex_list.append(vertex_name)
+
+    def add_out_supervertex(self, vertex_name: str):
+        self.out_supervertex_list.append(vertex_name)
+
+
+class SuperEdge(object):
+    r"""
+    The superedge structure in GripNet. Each superedge is a bipartite subgraph containing nodes from two categories
+    forming two node sets, connected by edges between them. The superedge can be regards as a heterogeneous graph
```

Comment:
The superedge --> A superedge

Reply:
Changed in [5c193a3](https://github.com/pykale/pykale/pull/338/commits/5c193a39c6e3d76aa9249ae5ff9546a17d6c60a2).

Reply:
Sorry, I mean the second "The superedge" should be "A superedge", not the beginning

Also "can be regards" --> "can be regarded"